sorting/InPlaceMergeSort.c

```diff
@@ -0,0 +1,119 @@
+#include<stdio.h>
+#include<stdlib.h>
+
+void swap(int * x, int * y )
+{
+	int z = *x;
+	*x = *y;
+	*y = z;
+}
+
+
+void insertionSort(int *arr, int s, int e)
+{
+   int i, key, j;
+   for (i = s+1; i < e; i++)
+   {
+       key = arr[i];
+       j = i-1;
+       while (j >= 0 && arr[j] > key)
+       {
+           arr[j+1] = arr[j];
+           j = j-1;
+       }
+       arr[j+1] = key;
+   }
+}
+
+
+
+void concat(int * arr, int l , int m, int r)
+{
+	int lh = m-l + 1 ,  rh = r-m ;
+	int * brr = malloc(sizeof(int)*lh) , * crr = malloc(sizeof(int)*rh) ;
+	int i, j = 0, k = l;
+	
+	insertionSort(arr, l , m);
+	
+	
+	insertionSort(arr,m+1,rh);
+	
+	printf("\n\n\n");
+	for(int i = l ; i < lh ; i++ )
+		printf("%d",arr[i]); 
```

Comment:
what is the purpose of this piece of code? If this is intended to print the left half of the array, it wouldn't print any elements for arrays of certain size.

Example: if the size of your array is 10, for the 2nd half of the array,
``l = 5, m = 7, r = 9``
``lh = m-l + 1 => 7-5 + 1 => 3``

since the for loops condition is ``i < lh``, it would fail, & not print anything.